### PR TITLE
Add cdflib

### PR DESCRIPTION
**Pull request by AlexS12**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by Juanlu001** [comment]
Why is the CI trying to install Python 2.7? :flushed: 

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=120930&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=431

**Comment by Juanlu001** [comment]
@conda-forge-admin, please rerender

**Comment by Juanlu001** [comment]
@conda-forge-admin, please restart ci

**Comment by Juanlu001** [comment]
@conda-forge/python sorry but I don't understand where does `No matching distribution found for twine>=1.11` come from... I can't see it listed as a requirement of cdflib or anywhere else

**Comment by AlexS12** [comment]
Me neither... I don't really now how to continue

File: recipes/cdflib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
```
Comment (npavlovikj) [inline recipes/cdflib/meta.yaml]: I don't know what part of the installation requires `twine`, but I would suggest you to add `twine >=1.11` in the `host` section.

File: recipes/cdflib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  run:
+    - python
```
Comment (npavlovikj) [inline recipes/cdflib/meta.yaml]: Looks like "cdflb" is Python 3, so please use `python >=3.5` here.

File: recipes/cdflib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  sha256: 9b027b0e5b7d84816aafa13edc5df3767c560881e8a93348f92cdfba29d7897a
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
```
Comment (npavlovikj) [inline recipes/cdflib/meta.yaml]: Please use `python >=3.5` here as well.

File: recipes/cdflib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    - pip
+  run:
+    - python
+    - numpy
+
+test:
+  imports:
+    - numpy
```
Comment (npavlovikj) [inline recipes/cdflib/meta.yaml]: You just need to test for the package import here, "cdflib".

File: recipes/cdflib/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    - cdflib
+
+about:
+  home: https://github.com/MAVENSDC/cdflib
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary:  'A python CDF reader toolkit'
+
```
Comment (npavlovikj) [inline recipes/cdflib/meta.yaml]: Please remove empty line.

File: recipes/cdflib/meta.yaml (hunk 2 above)
Comment (npavlovikj) [reply recipes/cdflib/meta.yaml]: Looks like you need to add `numpy` in `run` as well. In general, everything needed to run the package needs to go to the `run` section.

**Comment by npavlovikj** [comment]
@AlexS12 , the recipe LGTM. `noarch` recipes are only built on Linux, so we can ignore the AppVeyor error. Once @dstansby agrees to be listed as a maintainer for this recipe, we can go ahead and merge the recipe.

**Comment by npavlovikj** [comment]
> Why is the CI trying to install Python 2.7? 😳
> 
> https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=120930&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=431

As of now, Python packages are built for Python 2.7, Python 3.6 and Python 3.7, https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L562, unless otherwise specified.

**Comment by dstansby** [comment]
> @AlexS12 , the recipe LGTM. `noarch` recipes are only built on Linux, so we can ignore the AppVeyor error. Once @dstansby agrees to be listed as a maintainer for this recipe, we can go ahead and merge the recipe.

Yep, I'm happy 👍

**Comment by AlexS12** [comment]
> @AlexS12 , the recipe LGTM. `noarch` recipes are only built on Linux, so we can ignore the AppVeyor error. Once @dstansby agrees to be listed as a maintainer for this recipe, we can go ahead and merge the recipe.

@npavlovikj thanks for your review! Just a couple of things:
- what does LGTM mean?
- I have been able to use cdflib from windows when installed from pip. Does this mean that I won't be able to install it from conda?

**Comment by npavlovikj** [comment]
> > @AlexS12 , the recipe LGTM. `noarch` recipes are only built on Linux, so we can ignore the AppVeyor error. Once @dstansby agrees to be listed as a maintainer for this recipe, we can go ahead and merge the recipe.
> 
> @npavlovikj thanks for your review! Just a couple of things:
> 
> * what does LGTM mean?
> * I have been able to use cdflib from windows when installed from pip. Does this mean that I won't be able to install it from conda?

1. Sorry about the slang - LGTM means "Looks Good To Me" :)
2. You should be able to install `noarch` packages on Linux, OSX and Windows unless some of the run dependencies are missing. In this case, the only run dependencies are python and numpy, both available for all platforms. Thus, installing "cdflib" with conda on Windows should work just fine. In general, `noarch` packages can be built the same way on all platforms, and do not depend on the architecture of the OS, therefore `noarch`, https://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages. Since the build is same for all platforms, `noarch` packages save CI time by just being built on one platform instead, which is Linux here. Does this make sense?

**Comment by npavlovikj** [comment]
Also, thank you for contributing @AlexS12 🎉

**Pull request merged**